change name and mission of the WG

pchampin opened the pull request:
- This is a big PR (sorry).

I changed the name of the WG and deliverable to PUMPKIN (working title).

I mostly touched the first sections (Mission, Motivation & Background, Scope, Deliverables).
I refactored the text to make it more about the problem and less about the proposed solution. For the most, it is a reorganization of text that was already there, sometimes slightly rephrased, sometimes moved around (especially between the Motivation and Scope section).

The text that is really new compared to the previous version is highlighted in green to make it easier to review.

pchampin committed:
change name and mission of the WG

This is a big PR (sorry).

I changed the name of the WG and deliverable to PUMPKIN (working title).

I mostly touched the first sections (Mission, Motivation & Background,
Scope, Deliverables).
I refactored the text to make it more about the problem and less about
the proposed solution. For the most, it is a reorganization of text that
was already there, sometimes slightly rephrased, sometimes moved around
(especially between the Motivation and Scope section).

The text that is really new compared to the previous version is
highlighted in green to make it easier to review.

TallTed commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
-          <li>Interoperability between applications and various storage server implementations of the protocol.</li>
-          <li>A decentralized trust model enabling authentication and authorization mechanisms across the web.</li>
-        </ul>
+        <p><span class=new>A number of initiatives, such as <a href="https://indieweb.org/">IndieWeb</a>, <a href="https://www.w3.org/community/unhosted/">Unhosted</a> and <a href="https://solidproject.org/">Solid</a>, have emerged to propose alternative models for web applications.
```
```suggestion
        <p><span class=new>A number of initiatives — such as <a href="https://indieweb.org/">IndieWeb</a>, <a href="https://www.w3.org/community/unhosted/">Unhosted</a>, and <a href="https://solidproject.org/">Solid</a> — have emerged to propose alternative models for web applications.
```

TallTed commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
-          <li>A decentralized trust model enabling authentication and authorization mechanisms across the web.</li>
-        </ul>
+        <p><span class=new>A number of initiatives, such as <a href="https://indieweb.org/">IndieWeb</a>, <a href="https://www.w3.org/community/unhosted/">Unhosted</a> and <a href="https://solidproject.org/">Solid</a>, have emerged to propose alternative models for web applications.
+        They are</span> largely based on existing open web standards, and extend them to give priority to individual and community autonomy.</p>
```
"community" suggests a particular kind of group. "group" is more general.
```suggestion
        They are</span> largely based on existing open web standards, which they extend to give priority to individual and group autonomy.</p>
```

TallTed commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+        <p><span class=new>A number of initiatives, such as <a href="https://indieweb.org/">IndieWeb</a>, <a href="https://www.w3.org/community/unhosted/">Unhosted</a> and <a href="https://solidproject.org/">Solid</a>, have emerged to propose alternative models for web applications.
+        They are</span> largely based on existing open web standards, and extend them to give priority to individual and community autonomy.</p>
+        
+        <p>The <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> has been incubating <a href="https://solidproject.org/TR/">specifications</a> since 2018 towards that end: <q cite="https://www.w3.org/groups/cg/solid">to describe the interoperability between different classes of products by using web communication protocols, global identifiers, authentication and authoriza ...
```
```suggestion
        <p>The <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> has been incubating <a href="https://solidproject.org/TR/">specifications</a> since 2018 towards that end, i.e., <q cite="https://www.w3.org/groups/cg/solid">to describe the interoperability between different classes of products by using web communication protocols, global identifiers, authentication and authorization mechanisms, data formats and shapes, notifications, and query interfaces.</q>
```

TallTed commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+        They are</span> largely based on existing open web standards, and extend them to give priority to individual and community autonomy.</p>
+        
+        <p>The <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> has been incubating <a href="https://solidproject.org/TR/">specifications</a> since 2018 towards that end: <q cite="https://www.w3.org/groups/cg/solid">to describe the interoperability between different classes of products by using web communication protocols, global identifiers, authentication and authoriza ...
+        <span class=new>It has regularly sought cross-fertilization with the aforementioned communities and others (e.g. the <a href="https://www.w3.org/community/fed-id/">Federated Identity CG</a> and the <a href="https://www.w3.org/community/socialcg/">Social Web Incubator CG</a>).</span></p>
```
```suggestion
        <span class=new>It has regularly sought cross-fertilization with the aforementioned communities, among others (e.g., the <a href="https://www.w3.org/community/fed-id/">Federated Identity CG</a> and the <a href="https://www.w3.org/community/socialcg/">Social Web Incubator CG</a>).</span></p>
```

TallTed commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+        <span class=new>It has regularly sought cross-fertilization with the aforementioned communities and others (e.g. the <a href="https://www.w3.org/community/fed-id/">Federated Identity CG</a> and the <a href="https://www.w3.org/community/socialcg/">Social Web Incubator CG</a>).</span></p>
 
-        <p>The <cite><a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a></cite> has been incubating <a href="https://solidproject.org/TR/">Technical Reports</a> since 2018 towards that end: <q cite="https://www.w3.org/groups/cg/solid">to describe the interoperability between different classes of products by using web communication protocols, global identifiers, authenticati ...
+        <p>The wider Solid community has developed a <a href="https://solidproject.org/developers/tools/">suite of implementations, tools, and libraries</a> for developers, as well as <a href="https://solidproject.org/apps">applications</a> for users.:
```
```suggestion
        <p>The broader Solid community has developed a <a href="https://solidproject.org/developers/tools/">suite of implementations, tools, and libraries</a> for developers, as well as <a href="https://solidproject.org/apps">applications</a> for users.
```

TallTed commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+
         <p>
-                The design approach for the <a href="https://solidproject.org/TR/protocol">Solid Protocol</a> will continue substantial efforts undertaken previously in the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> to specify a set of interoperable protocol standards for enabling the use and development of Solid servers and applications across a <a href="https://g ...
+          <span class=new>The scope of the Working Group is to define a web protocol between client-side application, on the one hand, and storage and identity server, on the other hand, allowing</span> users to keep authority over their data, identity, and privacy.
```
```suggestion
          <span class=new>The scope of the Working Group is to define a web protocol for use between client-side applications, on the one hand, and storage and identity servers, on the other, allowing</span> users to keep authority over their data, identity, and privacy.
```

TallTed commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
         <p>
-                The design approach for the <a href="https://solidproject.org/TR/protocol">Solid Protocol</a> will continue substantial efforts undertaken previously in the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> to specify a set of interoperable protocol standards for enabling the use and development of Solid servers and applications across a <a href="https://g ...
+          <span class=new>The scope of the Working Group is to define a web protocol between client-side application, on the one hand, and storage and identity server, on the other hand, allowing</span> users to keep authority over their data, identity, and privacy.
+          This protocol must ensure interoperability between compliant applications and various storage and identity server,
```
```suggestion
          This protocol will ensure interoperability between compliant user applications, and compliant storage and identity servers,
```

TallTed commented on charter/index.html:
```diff
@@ -238,23 +244,20 @@ <h3>
-
                         <p>
-                          <b>Adopted Draft</b>: <a href="https://solidproject.org/ED/protocol">Solid Protocol, latest published version (0.10.0) or Editor's Draft</a>
+                            The PUMPKIN Protocol may include protocol details for integration with identity layers and mechanisms; access management and data integrity; notifications about resource changes; and, authorization mechanisms.
```
```suggestion
                            The PUMPKIN Protocol may include protocol details for integration with identity layers and mechanisms; access management and data integrity; notifications about resource changes; and authorization mechanisms.
```

TallTed commented on charter/index.html:
```diff
@@ -238,23 +244,20 @@ <h3>
                         </p>
                         <p>
-                            When possible, the Solid Protocol will evolve while maintaining a high degree of compatibility with existing implementations, of both servers and clients, and with features from prior versions. If incompatible changes have to be made, then they will be done by introducing a stage where both old and new protocols are supported, to allow the implementors to upgrade their ...
+                          <b>Input document</b>: <a href="https://solidproject.org/ED/protocol">Solid Protocol, latest published version (0.10.0) or Editor's Draft</a>, adopted from the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a>.
+                             <q lang="en-GB" cite="https://solidproject.org/TR/2022/protocol-20221231#introduction">An overarching design goal of the Solid ecosystem is to be evolvable and to provide fundamental affordances for decentralized web applications for information exchange in a way that is secure and privacy respecting. In this environment, actors allocate identifiers for their content, ...
```
```suggestion
                          <b>Input document</b>: <a href="https://solidproject.org/ED/protocol">Solid Protocol, latest published version (0.10.0 or later) or Editor's Draft</a>, adopted from the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a>.
                             <q lang="en-GB" cite="https://solidproject.org/TR/2022/protocol-20221231#introduction">An overarching design goal of the Solid ecosystem is to evolvably provide fundamental affordances for decentralized web applications for information exchange, in a way that is respects security and privacy. In this environment, actors allocate identifiers for their content; shape and store data wherever they have access; set access controls; and use their preferred applications and services to achieve their goals.</q>
```

pchampin replied:
this is a quote from the CG specification, so we should not change it

TallTed commented on charter/index.html:
```diff
@@ -267,7 +270,7 @@ <h3>
           <h3>Dependencies</h3>
 
-          <p>The proposed adopted draft for the Solid Protocol has a number of normative dependencies with a maturity level insufficient for a recommendation (Community Group reports, Editor's Drafts). The Working Group will need to decide how to deal with these dependencies in order to allow the Solid Protocol v1.0 to progress on the Recommendation track. This may include downgrading those depen ...
+          <p>The proposed adopted draft for the PUMPKIN Protocol has a number of normative dependencies with a maturity level insufficient for a recommendation (Community Group reports, Editor's Drafts). The Working Group will need to decide how to deal with these dependencies in order to allow the PUMPKIN Protocol v1.0 to progress on the Recommendation track. This may include downgrading those d ...
```
```suggestion
          <p>The draft of the PUMPKIN Protocol proposed for adoption has a number of normative dependencies with maturity levels insufficient for a recommendation (e.g., Community Group reports, Editor's Drafts). The Working Group will need to decide how to deal with these dependencies in order to allow the PUMPKIN Protocol v1.0 to progress on the Recommendation track. This may include downgrading some of these dependencies to non-normative references; specifying the required parts in the PUMPKIN Protocol itself; and/or relying on other specifications to fulfill the same purpose.</p>
```

TallTed commented on charter/index.html:
```diff
@@ -433,30 +435,30 @@ <h3 id="w3c-coordination">W3C Groups</h3>
 
             <dt><a href="https://www.w3.org/groups/wg/vc">Verifiable Credentials Working Group</a></dt>
-            <dd>To inquire on the suitability of credentials data models and serializations, and their potential integration with the Solid Protocol.</dd>
+            <dd>To inquire on the suitability of credentials data models and serializations, and their potential integration with the PUMPKIN Protocol.</dd>
```
```suggestion
            <dd>To inquire on the suitability of credential data models and serializations, and their potential integration with the PUMPKIN Protocol.</dd>
```

TallTed reviewed (changes requested):
Minor tweaks...

melvincarvalho commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
-          <li>Interoperability between applications and various storage server implementations of the protocol.</li>
-          <li>A decentralized trust model enabling authentication and authorization mechanisms across the web.</li>
-        </ul>
+        <p><span class=new>A number of initiatives, such as <a href="https://indieweb.org/">IndieWeb</a>, <a href="https://www.w3.org/community/unhosted/">Unhosted</a> and <a href="https://solidproject.org/">Solid</a>, have emerged to propose alternative models for web applications.
```
Why highlight these two?  What is the user base, how many DAU?  This looks weak, imho.

pchampin replied:
Feel free to suggest others if you think they are more relevant. Those were the ones that I was most aware of, and which I believe the Solid community has interacted with in the past.

melvincarvalho commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+
         <p>
-                The Working Group will:
+        Based on the <a href="https://github.com/solid/user-stories">user cases</a> identified by the Solid project, potentially extended by Working Group participants, the Working Group will:
```
The use cases are in a bad state, not maintained and woefully out of date.  Granted folks might not click through, but consider either making use cases more attractive, or removing this.

csarven replied:
The use cases are there to serve as input for discussions. They are not in a "bad state" but surely they can be always better organised or be more "attractive". It serves its purpose for the charter in that the CG has done some homework. Removing goes counter the main takeaway here.

melvincarvalho replied:
Do you think the use cases are in a good state?

https://github.com/solid/user-stories

If anyone clicks through, it gives a fairly unintuitive, wall of text, imho.

If anyone reading the charter works out that the user stories are in the issues tab, they will see this:

https://github.com/solid/user-stories/issues

At present time, it looks like a largely abandoned piece of work, with barely much activity over many years, and some people asking what's implemented without anyone having an answer.

It's OK if no one is going to look at it, but if someone does, would you consider it to be attractive, and well thought through?

melvincarvalho commented on charter/index.html:
```diff
@@ -238,23 +244,20 @@ <h3>
-                        <p>
-                            The Solid Protocol may include protocol details for integration with identity layers and mechanisms; access management and data integrity; notifications about resource changes; and, authorization mechanisms.
+                        <p class=new>
+                          The Working Group may consider other input, in addition to this specification, that is relevant to its scope and to the goal of this deliverable.
```
inputs

pchampin replied:
Yep, it was meant as "uncountable" input.

TallTed replied:
```suggestion
                          The Working Group will consider other input, in addition to this specification, that is relevant to its scope and to the goal of this deliverable.
```

melvincarvalho replied:
> Yep, it was meant as "uncountable" input.

Got it, thanks!

melvincarvalho commented on charter/index.html:
```diff
@@ -433,30 +435,30 @@ <h3 id="w3c-coordination">W3C Groups</h3>
 
             <dt><a href="https://www.w3.org/groups/wg/vc">Verifiable Credentials Working Group</a></dt>
-            <dd>To inquire on the suitability of credentials data models and serializations, and their potential integration with the Solid Protocol.</dd>
+            <dd>To inquire on the suitability of credentials data models and serializations, and their potential integration with the PUMPKIN Protocol.</dd>
```
Dont like this.  Taking on too much.  Could be done elsewhere.

pchampin replied:
Again, this is not a work item.

>  Could be done elsewhere.

But in order to decide whether additional work needs to be done, and where (could be the new WG, could be the VC WG, could be a CG...), we need to discuss.

melvincarvalho commented:
Generally like it.  3 comments:

- Consider another codename than PUMPKIN -- perhaps ask the mailing list?
- Have a compelling common understanding on Mission, as to what solid/pumpkin is, and what it does
- Consider removing some items, as I think it's taking on too much outside of the core, which will be hard enough to get to REC

TallTed commented:
For what it's worth, PUMPKIN has a distinguished history as a codename, having served the first RDFCore working group as a [codename for "node"](https://www.w3.org/2023/04/13-rdf-star-minutes.html#:~:text=in%20history%20of%20RDF%20development%2C%20we%20used%20%22pumpkin%22%20as%20a%20stand%2Din%20for%20%22nodes%22%2C%20which%20we%20hadn%27t%20figure%20out%20a%20name%20for%20yet) which didn't then have a proper name.

elf-pavlik commented:
> Consider another codename than PUMPKIN -- perhaps ask the mailing list?

As I understand, the whole purpose of using placeholder PUMPKIN is not to get sidetracked into bikesheding the name at this point.

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

csarven commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+        They are</span> largely based on existing open web standards, and extend them to give priority to individual and community autonomy.</p>
+        
+        <p>The <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> has been incubating <a href="https://solidproject.org/TR/">specifications</a> since 2018 towards that end, i.e., <q cite="https://www.w3.org/groups/cg/solid">to describe the interoperability between different classes of products by using web communication protocols, global identifiers, authentication and aut ...
+        <span class=new>It has regularly sought cross-fertilization with the aforementioned communities, among others (e.g., the <a href="https://www.w3.org/community/fed-id/">Federated Identity CG</a> and the <a href="https://www.w3.org/community/socialcg/">Social Web Incubator CG</a>).</span></p>
```
"Official" CG joint meetings included:

```suggestion
        <span class=new>It has regularly sought cross-fertilization with the aforementioned communities, among others (e.g., the <a href="https://www.w3.org/groups/cg/credentials/">Credentials Community Group</a>, the <a href="https://www.w3.org/groups/cg/wicg/">Web Platform Incubator Community Group</a>, the <a href="https://www.w3.org/groups/cg/socialcg/">Social Web Incubator CG</a>).</span></p>
```

pchampin replied:
thanks for the complement. Why remove the Federated Identity CG, though? I was pretty sure there was some interactions -- maybe not a dedicated joint meeting, though?

csarven replied:
Short: If you feel that mentioning the CG will be more appealing in this charter since they are also proposing a WG, go for it. I'd suggest to keep Credentials CG though.

Extended: Right. It was because of the initial joint meeting. If my memory serves me well, later that year some of the folks in WICG started FedID CG. No doubt that there are many individuals interacting randomly with various Groups. (You don't want to know where I've been :P )

elf-pavlik replied:
I started joining FedId meetings, and the work done there is relevant. There is also https://github.com/w3c/strategy/issues/427#issuecomment-1768348549
Solid-OIDC has an identity provider, so we do have an overlap there, which we should keep tracking.

TallTed replied:
```suggestion
        <span class=new>It has regularly sought cross-fertilization with the aforementioned communities, among others (e.g., the <a href="https://www.w3.org/groups/cg/credentials/">Credentials Community Group (CCG)</a>, the <a href="https://www.w3.org/community/fed-id/">Federated Identity Community Group</a>, the <a href="https://www.w3.org/groups/cg/socialcg/">Social Web Incubator Community Group</a>, and the <a href="https://www.w3.org/groups/cg/wicg/">Web Platform Incubator Community Group (WICG)</a>).</span></p>
```

samuelgoto replied:
We had @elf-pavlik and others come to the FedID CG a few times and we welcome the engagement here:

https://github.com/fedidcg/FedCM/issues/240#issuecomment-1782858001

I'm not sure if it is good or bad to make the association with the FedID CG for this group, but in case it helps, I think it would be fair to say that "it has regularly sought cross-fertilization with the FedID CG", based on this single thread I pointed to above.

csarven commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
 
-        <p>The <cite><a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a></cite> has been incubating <a href="https://solidproject.org/TR/">Technical Reports</a> since 2018 towards that end: <q cite="https://www.w3.org/groups/cg/solid">to describe the interoperability between different classes of products by using web communication protocols, global identifiers, authenticati ...
+        <p>The broader Solid community has developed a <a href="https://solidproject.org/developers/tools/">suite of implementations, tools, and libraries</a> for developers, as well as <a href="https://solidproject.org/apps">applications</a> for users.
+        <span class=new>See a separate <a href="https://timbl.solidcommunity.net/2023/03%20EWADA/TAGExplainerforSolid.html">explainer</a> for more detailed description of the Solid approach.</span>
```
As mentioned by the author elsewhere, this is a personal explainer. Is that what you meant by "separate"?

TallTed replied:
```suggestion
        <span class=new>Also see a personal <a href="https://timbl.solidcommunity.net/2023/03%20EWADA/TAGExplainerforSolid.html">explainer</a> for more detailed description of the Solid approach.</span>
```

cwilso replied:
(Side note: I don't think the adjective personal or separate is important here; you can just "see explainer")

TallTed replied:
tweaked to "Also see an explainer"...

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

csarven commented on charter/index.html:
```diff
@@ -238,23 +244,20 @@ <h3>
                         </p>
-                        <p>
-                            The Solid Protocol may include protocol details for integration with identity layers and mechanisms; access management and data integrity; notifications about resource changes; and, authorization mechanisms.
+                        <p class=new>
+                          The Working Group may consider other input, in addition to this specification, that is relevant to its scope and to the goal of this deliverable.
```
```suggestion
```

I think this paragraph is unnecessary (as hinted by Ted). Furthermore, the details on other input/proposals is already explained later in the text:

>The Working Group will adopt proposals that fulfill the requirements of <a href="https://www.w3.org/Guide/standards-track/#criteria">W3C Recommendation Track Readiness</a> [..]

pchampin replied:
In an ideal world, I agree it would be unnecessary. Given the concerns about rubber-stamping that have been expressed about the previous charter, I'm leaning towards keeping it. *Bis repetita placent*

TallTed replied:
@csarven 
> I think this paragraph is unnecessary (as hinted by Ted).

If I hinted at that, it was accidental, and should be disregarded. (I can find no such hint within this PR... Can you link me to it?)

csarven replied:
@pchampin Okie dokie!

@TallTed Perhaps I've misinterpreted you in https://github.com/solid/solid-wg-charter/pull/69/files#r1521554997 but I seem to share the same interpretation as @pchampin re uncountable. But you're right, you didn't hint "unnecessary" and I was jumping ahead with "uncountable" input, which I think is already covered by later text.

Any way... is it dinner time?

pchampin commented on charter/index.html:
```diff
@@ -412,19 +414,19 @@ <h2>Coordination</h2>
 
             <dt><a href="https://www.w3.org/groups/wg/did">Decentralized Identifier Working Group</a></dt>
-            <dd>To investigate the areas in which existing DIDs can be adopted by the Solid Protocol (and related technical reports), as well as to advance an understanding of whether a Solid-specific DID method is needed.</dd>
+            <dd>To investigate the areas in which existing DIDs can be adopted by the PUMPKIN Protocol (and related technical reports), as well as to advance an understanding of whether a specific DID method is needed.</dd>
```
> I think this should be out of scope

DIDs and WebID were developed largely independently, by different sub-communities, but they do overlap. And DID *is* a W3C recommendation, unlike WebID. So I don't think that the new WG can afford to simply ignore the DID WG.

Note that this does not says that the to-be-created REC will be using DIDs -- it just says that the two WG need to discuss it. The PUMPKIN WG may well come to the conclusion that, despite the overlap, DIDs are not appropriate and WebID is a better fit.

> DID is not one specification, but rather, over 100.

You could argue the same about URIs (https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml). Do you want to argue against the use of URIs as well?

>  Could you tell me if I am allowed to object

The design of the charter is not covered by the W3C process, but once the team proposes the charter to the AC, you may raise a formal objection: https://www.w3.org/2023/Process-20231103/#registering-objections

jyasskin replied:
How about:

```suggestion
            <dd>To investigate if existing DID methods should be adopted by the PUMPKIN Protocol (and related technical reports), as well as to advance an understanding of whether a specific DID method is needed.</dd>
```

(I don't see any pre-existing FOs that ask for this change, but it's always nice to improve consensus if possible.)

TallTed replied:
```suggestion
            <dd>To investigate whether existing DID methods should be adopted by the PUMPKIN Protocol (and related technical reports), and to advance an understanding of whether a specific DID method is needed.</dd>
```

melvincarvalho replied:
> You could argue the same about URIs (https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml). Do you want to argue against the use of URIs as well?

No I dont want to argue against URIs.  But I do want to argue against the heavy complexity involved with DID plus the 100+ sub specs.  It's adding a very large moving part that presents.  It puts Solid at risk, either needlessly, or for political reasons.  Nothing in solid right now requires DID.

> The design of the charter is not covered by the W3C process, but once the team proposes the charter to the AC, you may raise a formal objection: https://www.w3.org/2023/Process-20231103/#registering-objections

Consider this the first objection.  IMHO the spec would be much better without this extra moving part, that is already controversial, and opens a large can of worms.  This dependency could be done elsewhere, in the CG, or later after Solid is a REC.  IMHO that is the best chance of getting solid accross the line, rather than courting objection at step one.  But this all needs a longer write-up, if the addition of this item remains a hard dependency (and I understand it may be for political reasons) we can argue the case, hopefully in a clearer way that was argued with the original DID formal objections.

Strong -1

melvincarvalho replied:
> The PUMPKIN WG may well come to the conclusion that, despite the overlap, DIDs are not appropriate and WebID is a better fit.

This is a huge leap.  The two identity systems are not on equal footing in Solid.  Identity is at the heart of solid.

pchampin replied:
>  IMHO the spec would be much better without this extra moving part

The charter does not mandate those part to be in the spec; it merely states that the WG should consider it. I understand that you have a strong preference on the outcome of this discussion, but I don't think it is appropriate to prevent the WG from having it.

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin committed:
Update charter/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin commented on charter/index.html:
```diff
@@ -3,7 +3,7 @@
     <meta charset="utf-8"/>
 
-    <title>PROPOSED Solid Working Group Charter</title>
+    <title>PROPOSED PUMPKIN Working Group Charter</title>
```
It is not meant to be taken seriously, or to be taken at all. It is just a placeholder.

melvincarvalho replied:
OK, that wasnt all that clear, thanks for clarifying it.  Some folks in the wider community were a bit confused.

timbl replied:
The ecosystem do be built has to be a

- Personal - personal control of data
- Universal - in the sense you can use it for anything, not just social
- Manageable - in the sense that user has personal control of consent
- Protected -in the sense of resource-level access control
- Knowledge - in the sense of machine-readable facts
- Information 
- Network 

So let's call it PUMPKIN.  Oh, you already did.

csarven replied:
^ wins the interwebs.

melvincarvalho replied:
Please
use
My
Predicate
Knows
In
N3

cwilso replied:
In all seriousness - for the WG name, please don't use a codename, or a backronym.  Make the WG name be what scope the WG is intended to own, so it is instantly recognizable and understandable what the WG is supposed to be doing.

jyasskin reviewed (commented):
Minor comments:

TallTed commented on charter/index.html:
```diff
@@ -92,11 +96,12 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
-      <p class="mission">The <strong>mission</strong> of the <a href="">Solid Working Group <i class="todo">(LINK TBD)</i></a> is to standardize a protocol for user-centric data and its use of associated data interoperability and authentication schemes.
-      The resulting open standards intends to ensure interoperability and foster development in a decade-old ecosystem of application and server developers, aiming to return control of data back to users.</p>
+      <p class="mission new">The <strong>mission</strong> of the <a href="">PUMPKIN Working Group <i class="todo">(LINK TBD)</i></a> is to 
+      enable the development of web applications where data storage, authentication, and access control, are decoupled from the application provider.
```
```suggestion
      enable the development of web applications where data storage, entity authentication, access control, and application provider are all loosely coupled, as opposed to the web of today where these are typically all tightly coupled and changing one requires changing all, sometimes at the price of all past data.
```

melvincarvalho replied:
> FWIW, I think it's fine if the resulting protocol doesn't _guarantee_ decoupling, and the mission agrees with that by only promising that the protocol will _enable_ decoupling.

Is that really the mission?  Because there are existing examples of storage and identity loosely coupled on the web, such as using authentication headers.

What is left in the statement, is a some ways to add access control using Linked Data and Turtle which is loosely-coupled ie ACL and ACP.  Though this could be done with a standard policy too, e.g. everyone can read, owner can write.

This is helpful:

https://timbl.solidcommunity.net/2023/03%20EWADA/TAGExplainerforSolid.html

It says something about data portability (e.g. using Linked Data).  But if you consider the web and solid, the protocols can be packaged together, or individually.  This doesnt guarantee that all apps are client side (ultimately they wont be able to be at scale), for example nextcloud could perfectly use solid but not have client side apps.

A good clear mission would be quite helpful in the document.  For example explaining things that Solid can enable that are not already enabled on the web, such as leading towards a web operating system.  Or creating something on top of LDP to allow it to do more.  Or creating a first class HTTP Identity system, that is compliant with existing best practices.  I acknowledge that there will be multiple different missions of solid, but good to get a common understanding that is accurate.

TallTed commented on charter/index.html:
```diff
@@ -381,7 +383,7 @@ <h2>Success Criteria</h2>
 test suite of every feature defined in the specification.</p>
 	  <p>There should be testing plans for each specification, starting from the earliest drafts.</p>
-    <p>To promote interoperability, the group will publish <a href="https://solidproject.org/ED/qa">Solid QA</a> detailing testing policy, processes, and procedures.</p>
+    <p>To promote interoperability, the group will document a testing policy, processes, and procedures, taking <a href="https://solidproject.org/ED/qa">Solid QA</a> as an input.</p>
```
```suggestion
    <p>To promote interoperability, the group will document testing policies, processes, and procedures, taking <a href="https://solidproject.org/ED/qa">Solid QA</a> as an input.</p>
```

TallTed commented on charter/index.html:
```diff
@@ -412,19 +414,19 @@ <h2>Coordination</h2>
 
             <dt><a href="https://www.w3.org/groups/cg/n3-dev">Notation 3 (N3) Community Group</a></dt>
-            <dd>To investigate the possibility of transferring Solid Protocol's N3 Patch to the N3 CG where its standardisation development can continue with broader applicability.</dd>
+            <dd>To investigate the possibility of transferring PUMKIN Protocol's N3 Patch to the N3 CG where its standardization development can continue with broader applicability.</dd>
```
```suggestion
            <dd>To investigate the possibility of transferring PUMKIN Protocol's N3 Patch to the N3 Community Group where its standardization development can continue with broader applicability.</dd>
```

TallTed commented on charter/index.html:
```diff
@@ -433,30 +435,30 @@ <h3 id="w3c-coordination">W3C Groups</h3>
 
             <dt><a href="https://www.w3.org/groups/cg/unhosted">Unhosted Web Community Group</a></dt>
-            <dd>To explore the overlaps (CORS, user-centric storage, JSON-LD container descriptions) and differences (OAuth instead of WAC/ACP) between the Solid Protocol draft and their work on <a href="https://datatracker.ietf.org/doc/draft-dejong-remotestorage/">remoteStorage</a>.</dd>
+            <dd>To explore the overlaps (CORS, user-centric storage, JSON-LD container descriptions) and differences (OAuth instead of WAC/ACP) between the PUMPKIN Protocol draft and their work on <a href="https://datatracker.ietf.org/doc/draft-dejong-remotestorage/">remoteStorage</a>.</dd>
```
```suggestion
            <dd>To explore the overlaps (e.g., CORS, user-centric storage, JSON-LD container descriptions) and differences (e.g., OAuth instead of WAC/ACP) between the PUMPKIN Protocol draft and their work on <a href="https://datatracker.ietf.org/doc/draft-dejong-remotestorage/">remoteStorage</a>.</dd>
```

TallTed commented on charter/index.html:
```diff
@@ -433,30 +435,30 @@ <h3 id="w3c-coordination">W3C Groups</h3>
 
             <dt><a href="https://www.w3.org/groups/cg/webid">WebID Community Group</a></dt>
-            <dd>To provide integration feedback on the use of WebIDs in Solid Protocol (and related technical reports) in order to stabilise the fundamental notions and behaviors.</dd>
+            <dd>To provide integration feedback on the use of WebIDs in PUMPKIN Protocol (and related technical reports) in order to stabilize the fundamental notions and behaviors.</dd>
```
```suggestion
            <dd>To provide integration feedback on the use of WebIDs in PUMPKIN Protocol (and related technical reports) to stabilize the fundamental notions and behaviors.</dd>
```

pchampin replied:
I'd rather keep "in order to", which makes it easier for me to parse the relation between "to provide" and "to stabilize".

TallTed commented on charter/index.html:
```diff
@@ -267,7 +270,7 @@ <h3>
           <h3>Dependencies</h3>
 
-          <p>The proposed adopted draft for the Solid Protocol has a number of normative dependencies with a maturity level insufficient for a recommendation (Community Group reports, Editor's Drafts). The Working Group will need to decide how to deal with these dependencies in order to allow the Solid Protocol v1.0 to progress on the Recommendation track. This may include downgrading those depen ...
+          <p>The draft of the PUMPKIN Protocol proposed for adoption has a number of normative dependencies with maturity levels insufficient for a recommendation (e.g., Community Group reports, Editor's Drafts). The Working Group will need to decide how to deal with these dependencies in order to allow the PUMPKIN Protocol v1.0 to progress on the Recommendation track. This may include downgradin ...
```
```suggestion
          <p>The draft of the PUMPKIN Protocol proposed for adoption has a number of normative dependencies with maturity levels insufficient for a recommendation (e.g., Community Group reports, Editor's Drafts). To allow the PUMPKIN Protocol v1.0 to progress on the Recommendation track, the Working Group will need to decide how to deal with these dependencies. This may include downgrading some of these dependencies to non-normative references; specifying the required parts in the PUMPKIN Protocol itself; and/or relying on other specifications to fulfill the same purpose.</p>
```
```suggestion
          <p>The draft of the PUMPKIN Protocol proposed for adoption has a number of normative dependencies with maturity levels insufficient for a recommendation (e.g., Community Group reports, Editor's Drafts). The Working Group will need to decide how to deal with these dependencies in order to allow the PUMPKIN Protocol v1.0 to progress on the Recommendation track. This may include downgrading some of these dependencies to non-normative references; specifying the required parts in the PUMPKIN Protocol itself; and/or relying on other specifications to fulfill the same purpose.</p>
```

melvincarvalho commented:
> As I understand, the whole purpose of using placeholder PUMPKIN is not to get sidetracked into bikesheding the name at this point.

I can live with pumpkin, but it's raised a few eyebrows, and caused confusion because it sounds, imho, a bit unserious.

I could be wrong in that evaluation.  Names can be important as they project an image, and choosing a name for something CAN be bike-shedding, but isnt always bike-shedding

Let me give an example response from someone that has followed Solid casually for a long time:

> Obviously not the Cinderella story they expected

Some possible suggestions:

**ACORN**: Reflecting potential and growth, this name suggests the beginnings of something significant and expansive, much like the objectives of the SOLID project.

**OAK**: Symbolizing strength and stability, "OAK" could represent the robustness and reliability you aim for in your new spec.

**IVY**: Signifying growth and connection, "IVY" could represent the interlinked data and relationships facilitated by SOLID principles.

**BEACON**: Indicating guidance and illumination, this name suggests the spec's role in navigating new directions in web development.

**VERTEX**: Representing a peak or the highest point, "VERTEX" could symbolize the ambition of the SOLID project to achieve new heights in web standards and interoperability.

**ORIGIN**: Highlighting the foundational aspect of the project, this name suggests a new beginning or source from which further development can grow.

**HORIZON**: Signifying the exploration of new frontiers, "HORIZON" reflects the forward-looking nature of the SOLID project and its commitment to innovation.

Anyway I'll leave this as food for thought.  I dont know how long PUMPKIN is going to stick around, but perhaps worth a quick thread on the mailing list.

samuelgoto commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+        <p><span class=new>A number of initiatives — such as <a href="https://indieweb.org/">IndieWeb</a>, <a href="https://www.w3.org/community/unhosted/">Unhosted</a>, and <a href="https://solidproject.org/">Solid</a> — have emerged to propose alternative models for web applications.
+        They are</span> largely based on existing open web standards, which they extend to give priority to individual and group autonomy.</p>
+        
+        <p>The <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> has been incubating <a href="https://solidproject.org/TR/">specifications</a> since 2018 towards that end, i.e., <q cite="https://www.w3.org/groups/cg/solid">to describe the interoperability between different classes of products by using web communication protocols, global identifiers, authentication and aut ...
```
"global identifiers" doesn't seem to be something that is worth being proud of about?

elf-pavlik replied:
Hi @samuelgoto, I would find it interesting to compare the use of WebID as a global identifier to the use of email addresses as a global identifier.  In OIDC, `email` scope is very commonly used, and  [FedCM has email as part of user info](https://fedidcg.github.io/FedCM/#dom-identityuserinfo-email).
Probably, we should pick a better venue for that comparison than this inline comment.

samuelgoto replied:
Email addresses can be directed/sharded, rather than global.

elf-pavlik replied:
People were looking into potentially similar approaches with WebId, such as https://kushaldas.in/categories/webid.html.
It would be great to document those practices in one place eventually. From my current standpoint, email addresses and WebIDs should be pretty comparable as identifiers. 
Ultimately, people should be able to do what fits their needs, while correlatable shouldn't be the only option. Many individuals, probably even more organizations, want to establish a strong web identity. Even within the dev community, many people try to register the same handle across many online services.
Thinkin' about what can be realistically addressed in this or one of the next PRs here, maybe we should at least mention that the proposed WG should address known issues with correlatable identifiers. /cc @michielbdejong

timbl replied:
I see people using IDs for Solid on a wide scale, between totally public like your linked in profile (I [wrote about recently](https://www.w3.org/DesignIssues/PersonalPublic.html) )  to single-use large random number IDs for whistle-blowing. In between, medium security IDs like alumni, club, and work emails, where some people know the correlation and some don't.  Having a variety of IDPs o different types helps that. 

It is important that this point in the Solid architecture is an extension pont so we can play with different authentication protocols, and support more than one at a time.

TallTed replied:
@samuelgoto — Note that there is no restriction that any agent (especially, human) may only have _n_ (which I think you read as _1_) ["global identifiers"](https://github.com/solid/solid-wg-charter/pull/69/files#r1529372222) (e.g., WebIDs) for themselves. I could have one WebID that I use for all my Rebel Alliance interactions, another that I use for all my Galactic Republic interactions, another that I use for all my Jedi interactions, and yet another that I use for all my Galactic Empire interactions. If and when I choose to announce that any two of these are co-referent to "me", I may choose to do so in one of those arenas, but remain distinct in the others. Key points are that the assertion/implication of coreference is my decision, and that none of these incorporate a mandated "phone home" except in the course of dereferencing that WebID in order to confirm the user's assertion of control therof — which is a known activity.

"Global identity" *can* be "something that is worth being proud of", especially as it allows pseudonymous activity that is interpreted only based on that activity, and not based on the fact that, e.g., the pseudonymous author is 12 years old. (See some discussion of this in OSC's Ender series, particularly e.g., _[Valentine Wiggin](https://en.wikipedia.org/wiki/Valentine_Wiggin), under her pseudonym [Demosthenes](https://en.wikipedia.org/wiki/Demosthenes), publishes_.)

TallTed replied:
@samuelgoto — Also note that WebIDs can also be "directed/sharded" (i.e., they can indirectly identify an agent-in-a-role, like "CEO of Company_X" or a group-that-acts-as-a-singular, like "BoD of Company_X" where any Board member can act on the Board's behalf).

samuelgoto replied:
Maybe you should call these "unique" rather than "global" then?

TallTed replied:
They're "global" because (HTTP/S-)URI-based identifiers (like WebID) offer dereferenceability as a feature, always being resolvable to descriptions of the identified agent/entity. This dereferenceability is available anywhere, anytime, to anyone — though there might be some dereferences which require authentication by the caller in order to get that description, and some which can only be completed when the caller is within a VPN or similar, etc.

As to "unique", that refers to the identified agent/entity (i.e., the identifier uniquely identifies an agent/entity. People often get confused, and think the WebID identifier cannot be lexically repeated, as with UUIDs/GUIDs), and is only guaranteed within a timestamp, because domain registrations change, among other things. Within that timestamp, however, they are intended to uniquely identify an entity/agent, such that anyone who refers to a given WebID (or similar) is understood to intend to mean the same entity/agent. 

Outside the bounds of the indeterminate timestamp, the agent/entity identified by the WebID (or similar) may differ — but the descriptions of the "earlier" and "later" entities/agents are expected to include sufficient detail — such as timestamps of description creation, modification, termination — that the dereferencer can handle each

timbl commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+
         <p>
-                The design approach for the <a href="https://solidproject.org/TR/protocol">Solid Protocol</a> will continue substantial efforts undertaken previously in the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> to specify a set of interoperable protocol standards for enabling the use and development of Solid servers and applications across a <a href="https://g ...
+          <span class=new>The scope of the Working Group is to define a web protocol for use between client-side applications, on the one hand, and storage and identity servers, on the other, allowing</span> users to keep authority over their data, identity, and privacy.
```
a) It is an internet protocol.  (what is a web protocol? an internet protocol based on HTTP? ok)  I have always used "internet protocol" defined as the syntax and semantics and sequence of communication between different hosts on the Internet.  HTTP, FTP are internet protocols Solid is a more powerful one based on HTTP.

b) The protocol design gives you value.  It provides things which are true if you follow it.  If you use HTTP, .. eg ..you never don't overwrite other people's changes to a resource, caches work, and so on.
If you use Solid correctly you get these properties. 
 - Users can use any app with any pod. 
 - Choice of app and choice of pod are independent. So two new markets.
 - Users can control who has access to their data.
 - Developers can write apps without having to change the code running on a server. 
 
and so on.  When we charter the group in a way the most important thing may be that we are chartering it to provide those beneficial properties of the Solid ecosystem built on top of the Solid Protocol as a new platform

elf-pavlik replied:
> Applications that use this protocol can more easily allow their users to keep authority over their data, identity, and privacy

I wouldn't formulate it this way. It is the user who allows and denies applications access to specific data at will. On the one hand, applications are the most crucial part responsible for UX, but on the other hand, they are the least trusted and privileged part of the ecosystem.
The original sentence captures it better: The protocol itself allows the user to retain authority over their data.

melvincarvalho replied:
> On the one hand, applications are the most crucial part responsible for UX, but on the other hand, they are the least trusted and privileged part of the ecosystem.
> The protocol itself allows the user to retain authority over their data.

The architecture of the ecosystem places pod providers at the apex of trust and privilege, following a **federated** model.  This is not theoretical, we have witnessed it over and over in the federated model.  

Subsequently, identity providers operate within a trusted third-party framework.  Again this has brought centralization power to large centralized players on the web (typically email providers, which is a reason that login with email URIs is pushed so aggressively, often to the exclusion of all else).

It's worth noting that the previous integration of WebID-TLS offered more user agency, allowing **sever-side** support without compromising functionality. The addition of OIDC, despite initial assurances of coexistence, unfortunately led to the quiet deactivation of WebID-TLS, **on the server side**, a move that seems to lack a technical rationale.   When the OIDC folks removed WebID-TLS from the client side, they quietly removed it from the server side, without any mandate to do so.  Turning back on WebID-TLS on the server side, which should never have been removed,  would help improve this situation.

This decision underscores the inherent risk of optionality in technology standards, where political dynamics may influence technical decisions, another reason why it's important to have a well defined identity system based on web protocols ie HTTP URIs, at least at this point, and not open the door to optionality in identity, in the WG, which can quickly become political.

In short, despite the wording in the charter, the user is in _3rd place_ in terms of agency.  The pod owner is first, and has complete authority, including to cancel any user.  The identity provider then has power over the user, can impersonate the user, can silently track all user activity.  After that the user can add permissioned access control.  This is very useful and can lead to a web operating system, but that's not quite what's communicated in the charter, at this point.

TallTed replied:
> login with email URIs

Doesn't really exist. It's login with email *addresses*. Yes, those addresses are trivially converted to URIs (by prepending the `mailto:` scheme), but without that conversion, they remain addresses. Referring to them as "email URIs" does no-one any favors.

cwilso replied:
I think my biggest concern here is buried in "if you use Solid correctly".  That's not just if the user uses Solid correctly - it's relying on all app developers and other participants in the ecosystem to honor an (implicit?) pattern.

melvincarvalho replied:
> I think my biggest concern here is buried in "if you use Solid correctly". That's not just if the user uses Solid correctly - it's relying on all app developers and other participants in the ecosystem to honor an (implicit?) pattern.

Excellent point.  If the original web was "used correctly" many of the pervasive issues we see today would not exist.  

> Choice of app and choice of pod are independent. So two new markets.

Solid's app model, right now, is not incentivized.  Adverts can be blocked.  What is the incentive to use solid "correctly" for app makers?  The incentives are more aligned to couple apps and server together, and track users.  As someone that works with advanced client side apps every day, I can say it's extremely challenging for any apps to get any sustainable revenue.  In a competitive app market most of the apps will not cover their costs, leading to some dominant players, if at all.  Even more importantly there is the "king maker" effect.  In short, there are incentives missing for solid to be "used correctly".  As the app ecosystem matures, this will become more apparent.  For example how would you do advertising on a solid app, when it's easy to remove?

I think what's missing in the list is not the loose coupling of apps (this exists today elsewhere), but rather the *power* of apps working together on client and server side.  I think the idea of a web operating system could be explained more.

pchampin replied:
@jyasskin for the sake of usability, I'll turn this suggestion (+ the following thread) into a separate PR.

TallTed commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+        <p><span class=new>A number of initiatives — such as <a href="https://indieweb.org/">IndieWeb</a>, <a href="https://www.w3.org/community/unhosted/">Unhosted</a>, and <a href="https://solidproject.org/">Solid</a> — have emerged to propose alternative models for web applications.
+        They are</span> largely based on existing open web standards, which they extend to give priority to individual and group autonomy.</p>
+        
+        <p>The <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> has been incubating <a href="https://solidproject.org/TR/">specifications</a> since 2018 towards that end, i.e., <q cite="https://www.w3.org/groups/cg/solid">to describe the interoperability between different classes of products by using web communication protocols, global identifiers, authentication and aut ...
```
@jyasskin — Do you have a recommendation for replacement markup? (I don't see any such in the issues you linked.) That will make it much easier to act on your comment.

jyasskin replied:
Just use quotation marks.
```suggestion
        <p>The <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> has been incubating <a href="https://solidproject.org/TR/">specifications</a> since 2018 towards that end, i.e., "to describe the interoperability between different classes of products by using web communication protocols, global identifiers, authentication and authorization mechanisms, data formats and shapes, notifications, and query interfaces."
```

The `cite` URL is no longer directly attached to the quotation, but that URL wasn't available to users outside of browser devtools anyway. The citation is still available through the link on "W3C Solid Community Group" at the beginning of the paragraph.

csarven replied:
Right, "quotation marks" is a near alternative, but there is nothing wrong with `<q cite` here either. `q` is still recomended in HTML5 for exactly this purpose and the `cite` attribute is used as per https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-q-element . Are there recent accessibility findings that recommends against the use of `q` or its deprecation in HTML5?

No objection to using just quotes... but prefer precision with `<q cite`.

pchampin replied:
I share @csarven's taste for semantic markup, but I was convinced by @jyasskin's arguments about usability, discussed [here](https://github.com/whatwg/html/issues/10216).

cwilso commented on charter/index.html:
```diff
@@ -238,23 +244,20 @@ <h3>
           </p>
                 <dl>
-                    <dt id="solid-protocol" class="spec">Solid Protocol v1.0</dt>
+                    <dt id="PUMPKIN-protocol" class="spec">PUMPKIN Protocol v1.0</dt>
```
This is the place where it might make sense to continue to use "Solid".

cwilso replied:
Or, even better - just name it "a protocol specification", name Solid as input below, and then you can easily absorb it and call your protocol spec Solid without raising hackles over pre-disposing toward it now.

csarven replied:
@cwilso, exactly. I think that's currently covered in this PR in that the group proposes to work on a protocol where Solid Protocol (ED) will serve as one possible "input document" (see paragraph below in the charter). Background: https://github.com/solid/solid-wg-charter/pull/63#issuecomment-1931796745 based on past work/experience.

"PUMPKIN Protocol" or "a protocol specification" as per your suggest both work IMO.

Aside: Now, the reality is that while the Solid Protocol may be the leading input at this point in time, the charter is mindful of https://www.w3.org/Guide/standards-track/#criteria and so other works qualifying as such would be welcome as input towards "a protocol specification". For instance, while the Fedora API Specification 1.0 ( https://fcrepo.github.io/fcrepo-specification/ ) wasn't incubated in a W3C CG, it emerged from an open process with a reasonable dose of "strong and broad consensus" and implementations that are in use today. IMO, such works (that are also compatible with the Solid Protocol, e.g., LDP, and in fact using Web Access Control: https://solid.github.io/web-access-control-spec/ ) would qualify as input, but that's for the potential WG to decide.

cwilso replied:
I am at a bit of a loss to understand what the Fedora spec *IS*, exactly, in standards terms.  It claims to be a Recommendation/PR/CR, but it is not in any W3C group that I can tell? 

My net point here was that the WG might have a deliverable that ends up being called Solid, particularly presuming the WG chooses to adopt the current Solid protocol and use it.  Your Working Group should NOT be called Solid, because it doesn't mean anything, and you don't have that community together already to make such a decision.

acoburn replied:
@cwilso the Fedora Specification is a self-published document that comes from an organization called [Lyrasis](https://www.lyrasis.org/) and not part of any official standards body (e.g. W3C, IETF). I was involved in that work ~7 years ago. The hope at the time was that it would result in multiple, interoperable implementations of the Fedora API specification, but that never really materialized. AFAIK, there is only a [single implementation](https://fedora.lyrasis.org/), but I have also not been involved in quite some time.

The Fedora API is an interesting case in that it extends the LDP specification by adding features that are important to a particular domain (digital repositories). Those features include access control, notifications and resource versioning, which happen to overlap with the current Solid protocol. I think the point @csarven is making here is that, like the current Solid API protocol, the Fedora API involved an open process outside of an official standards body.

To @cwilso's net point: the Fedora project is also an interesting case because the API specification _and_ the reference implementation had the same name: Fedora. That led to various types of confusion, and I'm not sure it helped the project overall.

csarven replied:
* The Fedora API and the Solid Protocol share similarities in that they both can serve as inputs toward the protocol specification outlined in the charter. It is understood that the list of input documents is uncountable and additional works could be acknowledged to support the goal.
* Needless to say, the Solid Protocol has been incubated within the W3C CG since 2018. Although the Fedora API was not incubated within W3C space, it originated from an open-source community and process. (Their use of the Rec/CR/PR terminology has no particular weight / meaning at W3C.)

@cwilso, at least the Solid CG (based on discussions on record) shares your view that the WG will be called whatever it needs to be. We've moved on from proposing the "Solid" WG, hence this PR. Furthermore, given that the Solid Protocol is just one input document, I wouldn't rule out the possibility that the protocol specification produced by the WG may call it something entirely different. That's the intention behind "PUMPKIN Protocol v1.0".

So, perhaps we can just clarify the placeholder names:

the "Smashing WG" and the "Pumpkins Protocol"

:)

cwilso replied:
@csarven Love it!  "W3C and the Infinite Sadness"

cwilso replied:
@acoburn Thanks for the elucidation.  I think I was confused because Fedora is presented at places as if it's a W3C REC-track document (PR, etc), but afaict, it's not ever been a work item in W3C (even in a CG Note, I think, but I could be wrong there).

melvincarvalho replied:
> @cwilso, at least the Solid CG (based on discussions on record) shares your view that the WG will be called whatever it needs to be. We've moved on from proposing the "Solid" WG, hence this PR. Furthermore, given that the Solid Protocol is just one input document, I wouldn't rule out the possibility that the protocol specification produced by the WG may call it something entirely different. That's the intention behind "PUMPKIN Protocol v1.0".

Pointer to this would be good.  Still trying to get my head round this.  So PUMPKIN is an umbrella to Solid and specs like solid?  Makes not a huge amount of sense to me, as it would be taking on more, and I think Solid would work best as a REC if it took on a core subset, rather than, pivot to a superset.  Some context would be helpful.

pchampin replied:
Resolving this; as csarven pointed out, it may be that the final name of the protocol is neither Solid nor the name of the working group.

csarven commented on charter/index.html:
```diff
@@ -238,23 +244,20 @@ <h3>
                         </p>
                         <p>
-                            When possible, the Solid Protocol will evolve while maintaining a high degree of compatibility with existing implementations, of both servers and clients, and with features from prior versions. If incompatible changes have to be made, then they will be done by introducing a stage where both old and new protocols are supported, to allow the implementors to upgrade their ...
+                          <b>Input document</b>: <a href="https://solidproject.org/ED/protocol">Solid Protocol, latest published version (0.10.0) or Editor's Draft</a>, adopted from the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a>.
```
Putting the following suggestion on the table following https://github.com/solid/solid-wg-charter/pull/69/files#r1533485709 with the understanding that it'd be useful to have on record genuine intentions of the charter proposal:

```suggestion
                          <b>Input documents</b>: <a href="https://solidproject.org/ED/protocol">Solid Protocol, latest published version (0.10.0) or Editor's Draft</a></cite>, adopted from the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a>, <a href="https://fcrepo.github.io/fcrepo-specification/">Fedora API Specification 1.0</a>.
```

For those that are unfamiliar with the Fedora API Specification, it is extends LDP and generally compatible with the Solid Protocol. It has a solution for various deliberations that were raised in the Solid Protocol. It also normatively references Web Access Control: https://solid.github.io/web-access-control-spec/ , which is also normatively references in the Solid Protocol. So, this is all *highly* useful and constructive input.

That said, whether the inclusion of this potential input needs to meet a certain criteria, that'd be good to note here, e.g., does it actually need the "approval" of the Fedora Commons/Project to be mentioned in the charter, or can the charter or the WG simply re-use/borrow/incorporate recommendations from the Fedora API Specification 1.0 (CC BY 4.0 licensed) towards "a protocol specification" - I don't see why not but the Solid CG and W3C Team/community can make that call and put it on record.

cwilso replied:
This seems like an unclear formatting:  it says "Solid spec, adopted from SolidCG, Fedora API spec".  I think you mean "Solid spec, adopted from Solid CG; Fedora API spec"?  or bullets, or something.

I think the biggest concern would be provenance of Fedora features and IPR coverage; not that such is impossible to resolve, just needs to be done.

pchampin replied:
For the sake of usability, I'll turn this suggestion into a separate PR.

kjetilk commented on charter/index.html:
```diff
@@ -167,40 +172,41 @@ <h1 id="title">PROPOSED Solid Working Group Charter</h1>
+
         <p>
-                The Working Group will:
+        Based on the <a href="https://github.com/solid/user-stories">user cases</a> identified by the Solid project, potentially extended by Working Group participants, the Working Group will:
```
```suggestion
        Based on the <a href="https://github.com/solid/user-stories">user stories</a> identified by the Solid project, potentially extended by Working Group participants, the Working Group will:
```

These are specifically not use case, they are *user stories* and should be understood as such. They haven't been curated the way they were intended, but it is vitally important to have a low-threshold input mechanism for people who are not deeply that can also be useful to the development process. And that's pretty hard to achieve.

Asking people to submit use cases is high threshold, and also quite a burden on people. Writing mailing list or forum posts is nice to get feedback and incubate an idea, but both is high threshold and may often be too much to read for developers, and so does not meet the purpose.

So, yes, the curation there hasn't happened and it should. Once a story can be done, the issue should be closed with an explanation (like a pointer to documentation) on how. But the big point remains, the general public needs a way to communicate what they could use into the project. The WG could of course design a different mechanism for that, but it has been a largely neglected part of standards development that has to change.

melvincarvalho commented:
@timbl the innovation of Solid is to make the web Turing Complete, but no one is saying that.  Once the web is Turing Complete, then, over the course of time, it can be pushed to its limits.  There is a natural tendency for technologies to be pushed to their limits over time.  I'm sure you know this, but it's not that well articulated.  By making the the web Turing complete, any application you can imagine can be built.  That's the innovation, and why it should go to WG and be a REC.  Cut out what's unnecessary, and keep this one magical property.

TallTed commented on charter/index.html:
```diff
@@ -313,15 +316,14 @@ <h3>
         <section id="timeline">
-          <h3>Timeline</h3>
-
+          <h3>Timeline</h3>d
```
```suggestion
          <h3>Timeline</h3>
```

pchampin committed:
Apply suggestions from code review

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin committed:
Apply suggestions from code review

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pchampin committed:
Update charter/index.html

See https://github.com/whatwg/html/issues/10216 for rationale.

Co-authored-by: Jeffrey Yasskin <jyasskin@gmail.com>

pchampin committed:
Update charter/index.html

Co-authored-by: Kjetil Kjernsmo <kjetil@kjernsmo.net>

pchampin committed:
Apply comment in code review

This is based on @TallTed's suggestion at https://github.com/solid/solid-wg-charter/pull/69/files#r1523947482
but clarify that it is the on the WG to decide whether an input is relevant or not.

pchampin commented on charter/index.html:
```diff
@@ -267,7 +270,7 @@ <h3>
+          <p>The draft of the PUMPKIN Protocol proposed for adoption has a number of normative dependencies with maturity levels insufficient for a recommendation (e.g., Community Group reports, Editor's Drafts). The Working Group will need to decide how to deal with these dependencies in order to allow the PUMPKIN Protocol v1.0 to progress on the Recommendation track. This may include downgradin ...
 
           <p>Depending on the Working Group progress, including consideration for <a href="https://www.w3.org/Consortium/Process/#adequate-implementation">
           adequate implementation experience</a>, the Group may also decide to adopt the following dependencies as input documents:</p>
```
@jyasskin thanks for this suggestion, which highlights a lack of clarity in the charter. In my view, (at least some of) the specs listed below are in scope of the Solid Protocol at large, so the group should have the ability to use them and include the result in the Solid Protocol deliverable (even if that means splitting the deliverable in several parts) without rechartering. Obviously, this is not clear enough. I'll merge this PR, then make a separate PR to clarify this particular point.

TallTed commented on charter/index.html:
```diff
@@ -238,23 +244,20 @@ <h3>
-                        <p>
-                            The Solid Protocol may include protocol details for integration with identity layers and mechanisms; access management and data integrity; notifications about resource changes; and, authorization mechanisms.
+                        <p class=new>
+                          The Working Group will consider other input, in addition to this specification, if it deems it relevant to its scope and to the goal of this deliverable.
```
Merged directly to this PR, without time to review

Too many "it(s)" with indeterminate antecedents.
```suggestion
                          The Working Group will consider other input, in addition to this specification, if the Working Group deems that input to be within the scope of its charter and relevant to the goals of its deliverables.
```
